transport/nio/src/main/java/org/kaazing/gateway/transport/nio/internal/NioHandshakeFilter.java

```diff
@@ -59,8 +59,8 @@ public void onPostAdd(IoFilterChain parent, String name, NextFilter nextFilter)
     }
 
     @Override
-    protected void doSessionOpened(NextFilter nextFilter, IoSessionEx session) throws Exception {
+    protected void doMessageReceived(NextFilter nextFilter, IoSessionEx session, Object message) throws Exception {
         taskExecutor.shutdownNow();
-        nextFilter.sessionOpened(session);
+        nextFilter.messageReceived(session, message);
```

Comment:
Is it ok if taskExecutor is called multiple times?  I assume it is...

Consider having this filter remove itself from the filter chain.  Probably not needed, but I'm more just curious